### PR TITLE
Fix `convert_to_na` with date

### LINKED ISSUE #289: Fix failing test on windows-devel
```r
── Error (test-convert_to_na.R:90:3): convert_to_na other classes ──────────────
Error in `charToDate(x)`: character string is not in a standard unambiguous format
Backtrace:
     ▆
  1. ├─testthat::expect_message(...) at test-convert_to_na.R:90:2
  2. │ └─testthat:::expect_condition_matching(...)
  3. │   └─testthat:::quasi_capture(...)
  4. │     ├─testthat (local) .capture(...)
  5. │     │ └─base::withCallingHandlers(...)
  6. │     └─rlang::eval_bare(quo_get_expr(.quo), quo_get_env(.quo))
  7. ├─datawizard::convert_to_na(d, na = list(3, "c", TRUE, "2022-01-02"))
  8. └─datawizard:::convert_to_na.data.frame(...) at datawizard/R/convert_to_na.R:43:2
  9.   └─base::lapply(...) at datawizard/R/convert_to_na.R:186:2
 10.     ├─datawizard (local) FUN(X[[i]], ...)
 11.     └─datawizard:::convert_to_na.Date(X[[i]], ...) at datawizard/R/convert_to_na.R:43:2
 12.       ├─base::which(x %in% as.Date(na)) at datawizard/R/convert_to_na.R:137:4
 13.       ├─x %in% as.Date(na) at datawizard/R/convert_to_na.R:137:4
 14.       ├─base::as.Date(na) at datawizard/R/convert_to_na.R:137:4
 15.       └─base::as.Date.character(na)
 16.         └─base (local) charToDate(x)
```

### PR DESCRIPTION
**Pull request by etiennebacher**
Close #289

### CODE REVIEW

File: tests/testthat/test-convert_to_na.R [hunk 1]
```diff
@@ -83,11 +91,15 @@ test_that("convert_to_na other classes", {
...
+    c = as.Date(c("2022-03-22", "2022-01-02", "2022-02-02", "2021-04-02", "2020-01-19")),
     d = c(NA, NA, FALSE, FALSE, NA),
     e = as.complex(1:5)
   )
-  expect_message(x <- convert_to_na(d, na = list(3, "c", TRUE, "2022-01-02")))
+  expect_message(
+    convert_to_na(d, na = list(3, "c", TRUE, "2022-01-02")),
+    "must be of class 'Date'"
+  )
+  expect_message(x <- convert_to_na(d, na = list(3, "c", TRUE, as.Date("2022-01-02"))))
```
Comment (IndrajeetPatil) [inline tests/testthat/test-convert_to_na.R:102]: Maybe use `expect_no_message()` to clarify the expectation?

**Review by IndrajeetPatil — APPROVED** [review]
LGTM. Thansk, @etiennebacher!

File: tests/testthat/test-convert_to_na.R (hunk 1 above)
Comment (IndrajeetPatil) [reply tests/testthat/test-convert_to_na.R:102]: https://testthat.r-lib.org/reference/expect_no_error.html

Comment (etiennebacher) [reply tests/testthat/test-convert_to_na.R:102]: this was actually a mistake, there shouldn't be an `expect_message()` here

**Pull request merged**